ehr/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dto/converter/PatientToDtoConverter.java

```diff
@@ -3,49 +3,105 @@
+import org.hl7.fhir.r4.model.StringType;
 import org.hl7.gravity.refimpl.sdohexchange.dto.response.PatientDto;
+import org.hl7.gravity.refimpl.sdohexchange.fhir.UsCorePatientExtensions;
 import org.hl7.gravity.refimpl.sdohexchange.util.FhirUtil;
 import org.springframework.core.convert.converter.Converter;
 
 import java.time.LocalDate;
 import java.time.Period;
 import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Optional;
 
 public class PatientToDtoConverter implements Converter<Patient, PatientDto> {
 
+  public static final String NOT_AVAILABLE = "N/A";
+
   @Override
   public PatientDto convert(Patient patient) {
     PatientDto patientDto = new PatientDto(patient.getIdElement()
         .getIdPart());
     patientDto.setName(patient.getNameFirstRep()
         .getNameAsSingleString());
-    patientDto.setGender(ObjectUtils.defaultIfNull(patient.getGender(), Enumerations.AdministrativeGender.UNKNOWN)
-        .getDisplay());
-
     //Get Date of Birth and Age
-    if (patient.getBirthDate() != null) {
+    if (patient.hasBirthDate()) {
       LocalDate dob = FhirUtil.toLocalDate(patient.getBirthDateElement());
       patientDto.setDob(dob);
       patientDto.setAge(Period.between(dob, LocalDate.now(ZoneOffset.UTC))
           .getYears());
     }
+    //Get gender
+    patientDto.setGender(ObjectUtils.defaultIfNull(patient.getGender(), Enumerations.AdministrativeGender.UNKNOWN)
+        .getDisplay());
+    //Get communication language
+    patientDto.setLanguage(patient.getCommunication()
+        .stream()
+        .filter(Patient.PatientCommunicationComponent::getPreferred)
+        .map(c -> c.getLanguage()
+            .getCodingFirstRep()
+            .getDisplay())
+        .findFirst()
+        .orElse(NOT_AVAILABLE));
     //Get Address full String. No need to compose it on UI side.
-    Address address = patient.getAddress()
+    patientDto.setAddress(patient.getAddress()
         .stream()
         .filter(a -> Address.AddressUse.HOME.equals(a.getUse()))
+        .map(this::convertAddress)
         .findFirst()
-        .orElse(null);
-    if (address != null) {
-      StringBuilder sb = new StringBuilder().append(address.hasLine() ? address.getLine()
-          .get(0) + "\n" : "");
-      sb.append(Strings.isNullOrEmpty(address.getCity()) ? "" : address.getCity() + ", ");
-      sb.append(Strings.isNullOrEmpty(address.getState()) ? "" : address.getState() + " ");
-      sb.append(Strings.isNullOrEmpty(address.getPostalCode()) ? "" : address.getPostalCode());
-      patientDto.setAddress(sb.toString());
-    }
+        .orElse(NOT_AVAILABLE));
 
+    List<ContactPoint> telecom = patient.getTelecom();
+    //TODO: Reimplement after clarification
+    //Get phone number
+    patientDto.setPhone(telecom.stream()
+        .filter(t -> ContactPoint.ContactPointSystem.PHONE.equals(t.getSystem()))
+        .map(ContactPoint::getValue)
+        .findFirst()
+        .orElse(NOT_AVAILABLE));
```

Comment:
Why not just return null and UI will will decide what to show, N/A or something else?

Reply:
Okay, will fix this.

Reply:
Fixed.